mozilla/addons-server
# Allow adding, viewing information about and deleting soft-blocks in the admin

Fixes https://github.com/mozilla/addons/issues/15013

### Description

Like other blocklist operations, everything goes through a `BlocklistSubmission`, which gains a new `block_type` field to determine what kind of block to apply for this particular submission. Users are not allowed to modify `Block`s directly.

Note: If a version is already soft-blocked or hard-blocked, that version won't be selectable when creating a new `BlocklistSubmission`, even of another type. However, the code underneath should handle overriding the block type for a given blocked version in a submission, and future actions will be added to soften or harden blocks.

### Testing

Play around with the admin and add/delete `BlocklistSubmission`, and then inspect the blocks until you're satisfied. Few things you can try to mix together:
- Make soft or hard blocks
- Make soft *and* hard blocks on add-ons with multiple versions (requires separate submissions)
- `BlocklistSubmission` for add-ons with `average_daily_users` above `settings.DUAL_SIGNOFF_AVERAGE_DAILY_USERS_THRESHOLD` are subject to approval as before. You can check that it still behaves as expected. You shouldn't be able to modify a block type for a submission you're approving (arbitrary decision on my part, we could change this later)
- Make soft and/or hard blocks for multiple guids at the same time, for add-ons containing multiple versions
- Etc
